Solve race condition in writer_handler and redefine msg data type

ethanec opened the pull request:
ezgo_arduino.ino:
1. Revise data structure

arduino_control.cpp:
1. Solve race condirion in writer_handler
2. Add some output msgs on terminal to check the state of system
3. Redefine msg data type

ethanec committed:
Revise document title

ethanec committed:
Merge branch 'master' of https://github.com/ethanec/ezgo_arduino

ethanec committed:
Solve race condition in writer_handler and redefine msg data type

ezgo_arduino.ino:
1. Revise data structure

arduino_control.cpp:
1. Solve race condirion in writer_handler
2. Add some output msgs on terminal to check the state of system
3. Redefine msg data type

Chao-Shun-Cheng commented:
Please check the coding style (follow linux coding style) and remove unnecessary sections (e.g. print for checking) 

By the way, this is nice work. Good job !

ethanec committed:
Update files and project topic

1. Add .xlsx file that defines canbus data field
2. Commit some testing codes
3. Change project topic from "ezgo_control" to "ezgo_arduino"

Chao-Shun-Cheng commented:
@ethanec The package name should be "ezgo_control" which includes arduino control node and ECU control node.

Chao-Shun-Cheng commented:
> @ethanec The package name should be "ezgo_control" which includes arduino control node and ECU control node.

Until now, has arduino control been achieved including delay issue ?
